communitiesuk/funding-service-design-assessment
# Fs 4268 fund summary logic

### Change description
Reviewed the logic of the Assessment Tool Dashboard (Landing) page to simplify and make it easier to debug, and easier to test.
- Removed the `FORCE_OPEN_ALL_LIVE_ASSESSMENT_ROUNDS` env var as we can mimic this with dates in the DB
- Extracted some of the logic in the fund_summary template into macros for easier testing
- Renamed fund_summary.py to round_summary.py as it was referred to by both names in different places, and is actually used per round not per fund
- Resolves FS-4268 (link display for EOI)
- Extracted the logic around what status a round is currently in to a single dedicated method and class (`RoundStatus`) to hold the results. Refactored code to use this.
- Wrote unit tests for the above



### How to test
Launch the assessment tool and check the round summaries display correctly for rounds in different statuses


### Screenshots of UI changes (if applicable)
Only UI change is the fixing of the link text for EOI (FS-4268)
![Screenshot 2024-03-14 at 13 30 13](https://github.com/communitiesuk/funding-service-design-assessment/assets/1729216/d22afad3-974f-4625-83ec-6c296acdce30)
